projects/vdk-plugins/vdk-structlog/src/vdk/plugin/structlog/structlog_plugin.py

```diff
@@ -0,0 +1,171 @@
+from vdk.api.plugin.hook_markers import hookimpl
+from vdk.api.plugin.plugin_registry import HookCallResult
+from vdk.api.plugin.plugin_registry import IPluginRegistry
+from vdk.internal.builtin_plugins.run.execution_results import ExecutionResult
+from vdk.internal.builtin_plugins.run.execution_results import StepResult
+from vdk.internal.builtin_plugins.run.job_context import JobContext
+from vdk.internal.builtin_plugins.run.step import Step
+from vdk.internal.core.config import ConfigurationBuilder
+from vdk.internal.core.context import CoreContext
+from vdk.plugin.structlog.constants import STRUCTLOG_LOGGING_FORMAT_KEY
+from vdk.plugin.structlog.constants import STRUCTLOG_LOGGING_METADATA_ALL_KEYS
+from vdk.plugin.structlog.constants import STRUCTLOG_LOGGING_METADATA_DEFAULT
+from vdk.plugin.structlog.constants import STRUCTLOG_LOGGING_METADATA_KEY
+from vdk.plugin.structlog.filters import AttributeAdder
+from vdk.plugin.structlog.formatters import create_formatter
+
+"""
+
+
+Handlers
+https://docs.python.org/3/library/logging.html#handler-objects
+https://docs.python.org/3/howto/logging-cookbook.html#imparting-contextual-information-in-handlers
+
+Filters
+https://docs.python.org/3/library/logging.html#filter-objects
+
+LogRecords
+https://docs.python.org/3/library/logging.html#logrecord-objects
+
+Additional reading
+
+https://docs.python.org/3/howto/logging.html
+https://realpython.com/python-logging/
+https://realpython.com/python-logging-source-code/
+
+Logging adapters a.k.a. bound loggers
+https://docs.python.org/3/howto/logging-cookbook.html#context-info
+
+Example:
+
+class BoundLogger(logging.LoggerAdapter):
+    def process(self, msg, kwargs):
+        # merge bound extra dict with existing exra dict if any
+        if "extra" in kwargs:
+            kwargs["extra"] = {**self.extra, **kwargs["extra"]}
+        else:
+            kwargs["extra"] = self.extra
+        return msg, kwargs
+
+
+log = logging.getLogger(__name__)
+bound = BoundLogger(log, {"bound_key": "bound_value"})
+bound.warning("From vdk_initialize", extra={"first": "first_value", "second": "second_value"})
+bound.warning("Something else from vdk_initialize")
+
+"""
+
+
+class StructlogPlugin:
+    @hookimpl(tryfirst=True)
+    def vdk_configure(self, config_builder: ConfigurationBuilder):
+        config_builder.add(
+            key=STRUCTLOG_LOGGING_METADATA_KEY,
+            default_value=",".join(list(STRUCTLOG_LOGGING_METADATA_DEFAULT.keys())),
+            description=(
+                f"Possible values: {STRUCTLOG_LOGGING_METADATA_ALL_KEYS}"
+                "User-defined key-value pairs added to the logger's context will be displayed after the metadata, but before the message"
+            ),
+        )
+
+        config_builder.add(
+            key=STRUCTLOG_LOGGING_FORMAT_KEY,
+            default_value=",".join(list(STRUCTLOG_LOGGING_METADATA_DEFAULT.keys())),
+            description=(f"Possible values: json, console"),
```

Comment:
Shouldn't the possible values be a configurable list?

Reply:
Also there's no actual description of the config var, just the possible values

Reply:
This applies to the other config option, I think. I tried using a list, but it always parses a string for some reason. If you do something like

```python
config_builder.add(
    key="my_key",
    default_value=["value1", "value2", "value3"],
    description="Some description",
```

and then in config.ini

```
my_key=["something", "something_else"]
```

You'll actually get

```python
['[','s','o','m', etc...']']
```

Reply:
sorry, I meant that the possible values shouldn't be hard coded in here, instead they should be pulled from somewhere else

Reply:
Done.